Wazuh agent 4.4.0 doesn't compile in Alpine Linux i386, ARM64, ARMv6 and ppc64le

As part of [the central issue of supporting Alpine,](https://github.com/wazuh/wazuh-jenkins/issues/3994) it's necessary to support i386, ARM64, ARMv6, and ppc64le architecture with Alpine OS. 

Currently, the Wazuh agent doesn't compile: 

**x86 (i386)**
```
checking zlib.h usability...     CC external/msgpack/src/zone.o
no
checking zlib.h presence... no
checking for zlib.h... no
checking for gzopen in -lz... no
checking bzlib.h usability... no
checking bzlib.h presence... no
checking for bzlib.h... no
checking for libbz2... make depend && make _build_libs
make[2]: Entering directory '/build/SPECS/wazuh-agent/src/wazuh-wazuh-840f754/src/external/openssl'
no
mkdir -p external/popt/build/ && cd external/popt/build/ && cmake ..
checking for the pthreads library -lpthreads... no
checking whether pthreads work without any flags... yes
checking for joinable pthread attribute... make[2]: Leaving directory '/build/SPECS/wazuh-agent/src/wazuh-wazuh-840f754/src/external/openssl'
-- The C compiler identification is GNU 11.2.1
PTHREAD_CREATE_JOINABLE
checking if more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... make[2]: Entering directory '/build/SPECS/wazuh-agent/src/wazuh-wazuh-840f754/src/external/openssl'
gcc  -I. -Iinclude -fPIC -pthread -mx32 -Wa,--noexecstack -Os -fomit-frame-pointer -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib/engines-1.1\"" -DNDEBUG -Os -fomit-frame-pointer -MMD -MF apps/app_rand.d.tmp -MT apps/app_rand.o -c -o apps/app_rand.o apps/app_rand.c
cc1: sorry, unimplemented: 64-bit mode not compiled in
make[2]: *** [Makefile:679: apps/app_rand.o] Error 1
make[2]: Leaving directory '/build/SPECS/wazuh-agent/src/wazuh-wazuh-840f754/src/external/openssl'
make[1]: *** [Makefile:177: build_libs] Error 2
make[1]: Leaving directory '/build/SPECS/wazuh-agent/src/wazuh-wazuh-840f754/src/external/openssl'
make: *** [Makefile:919: external/openssl/libssl.a] Error 2
make: *** Waiting for unfinished jobs....
yes
```

**arm64**
```
    CC rootcheck/config.o
    CC rootcheck/os_string.o
rootcheck/os_string.c:73:10: fatal error: linux/a.out.h: No such file or directory
   73 | #include <linux/a.out.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:1970: rootcheck/os_string.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/build/SPECS/wazuh-agent/src/wazuh-wazuh-840f754/src'
make: *** [Makefile:753: agent] Error 2
+ die 'build failed'
```

--- a/src/rootcheck/os_string.c
+++ b/src/rootcheck/os_string.c
@@ -45,7 +45,7 @@
 
 #ifdef SOLARIS
 #include <sys/exechdr.h>
-#elif defined Darwin || defined HPUX
+#elif defined Darwin || defined HPUX || defined ALPINE
 
 /* For some reason darwin does not have that */
 struct exec {
@@ -69,11 +69,7 @@ struct exec {
 
 #else
 
-#ifdef ALPINE
-#include <linux/a.out.h>
-#else
 #include <a.out.h>
-#endif
 
 #endif
 

